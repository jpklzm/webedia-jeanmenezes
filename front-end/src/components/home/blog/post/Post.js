import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.scss';

export const FeaturedPost = ({ post }) => {
  // Fullsized post with all data.
  return (
    <a href="#">
      <div className={styles.featuredCard}>
        <div className={styles.featuredContainer}>
          <h1 className={styles.featuredTitle}>{post.title}</h1>
          <h2 className={styles.description}>{post.description}</h2>
          <img src={post.imageUrl} className={styles.featuredImage} />
          <div className={styles.contentArea}>
            <hr className={styles.hr} />
            <h1 className={styles.subTitle}>{post.subTitle}</h1>
            <p>{post.content}</p>
          </div>
        </div>
      </div>
    </a>
  );
};

export const Post = ({ post }) => {
  // Small post with reduced data for two columns layout
  return (
    <div className="col-sm-6">
      <a href="#">
        <div className={styles.card}>
          <div className={styles.container}>
            <h1 className={styles.title}>{post.title}</h1>
            <img src={post.imageUrl} className={styles.image} />
          </div>
        </div>
      </a>
    </div>
  );
};

const postPropTypes = {
  post: PropTypes.object,
};

FeaturedPost.propTypes = postPropTypes;
Post.propTypes = postPropTypes;
