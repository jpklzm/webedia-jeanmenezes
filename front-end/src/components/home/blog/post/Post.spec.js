import React from 'react';
import renderer from 'react-test-renderer';

import { Post } from './Post';

const postMock = {
  title: 'title',
  subTitle: 'subTitle',
  description: 'description',
  content: 'content',
  imageUrl: 'http://www.fragil.org/wp-content/themes/courage/assets/images/placeholder_featured_image.svg',
};

test('Snapshot', () => {
  const component = renderer.create(<Post post={postMock} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
