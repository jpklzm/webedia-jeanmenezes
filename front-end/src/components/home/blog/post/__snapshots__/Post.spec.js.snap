// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div
  className="col-sm-6"
>
  <a
    href="#"
  >
    <div
      className="card"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          title
        </h1>
        <img
          className="image"
          src="http://www.fragil.org/wp-content/themes/courage/assets/images/placeholder_featured_image.svg"
        />
      </div>
    </div>
  </a>
</div>
`;
